maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/SurefireHelper.java

```diff
@@ -142,7 +142,9 @@ public static void reportExecution( SurefireReportParameters reportParameters, R
                                         PluginConsoleLogger log, Exception firstForkException )
         throws MojoFailureException, MojoExecutionException
     {
-        if ( firstForkException == null && !result.isTimeout() && result.isErrorFree() )
+        if ( firstForkException == null && !result.isTimeout() && result.isErrorFree()
```

Comment:
@oehme 

My proposal:
```
boolean isError = firstForkException != null || result.isTimeout() || !result.isErrorFree();
boolean isFlaky = reportParameters.getFailOnFlakeCount() > 0
                && result.getFlakes() >= reportParameters.getFailOnFlakeCount();
if ( isError || isFlaky )
{
    ...
}
else
{
    if ( isError )
    {
        msg.append( "There are test failures." );
    }

    if ( isFlaky )
    {
        msg.append( "There are too many flakes." );
    }
    
    msg.append( "\n\nPlease refer to " );
}
```

Reply:
> 
> 
> Got it, thanks!
> 
> So is this option something you'd consider adding? In that case I'd flesh out the PR some more.

Yes I would consider this PR adding to Surefire.
Scary of the future PRs a bit ;-)